optimum_benchmark/backends/base.py

```diff
@@ -68,7 +68,13 @@ def __init__(self, config: BackendConfigT):
             self.automodel_loader = get_timm_automodel_loader()
             self.pretrained_processor = None
             self.generation_config = None
-
+        elif self.config.library == "llama_cpp":
+            self.logger.info("\t+ Benchmarking a Llama.cpp model")
+            self.pretrained_config = None
+            self.model_shapes = {}
+            self.model_type = self.config.task
```

Comment:
I think it's better to use the model type info in https://huggingface.co/TheBloke/TinyLlama-1.1B-Chat-v1.0-GGUF/blob/main/config.json gguf repos usually have this or a proper transformers config https://huggingface.co/second-state/Mistral-Nemo-Instruct-2407-GGUF/blob/main/config.json

Reply:
I change it to now try to load the transformers config directly. The transformers loading default to model type info if the config is not a proper transformer config so both case should be handled now